src/lib/y2system_role_handlers/worker_role_finish.rb

```diff
@@ -21,28 +21,57 @@
 
 require "yast"
 require "y2caasp/cfa/minion_master_conf"
+require "y2caasp/cfa/systemd_timesyncd"
 
 module Y2SystemRoleHandlers
   # Implement finish handler for the "worker" role
   class WorkerRoleFinish
     include Yast::Logger
 
-    # Configure Salt minion
     def run
       role = ::Installation::SystemRole.find("worker_role")
-      master_conf = ::Y2Caasp::CFA::MinionMasterConf.new
       master = role["controller_node"]
+      log.info("The controller node for this worker role is: #{master}")
+
+      configure_salt_minion(master)
+      configure_systemd_timesync(master)
+      enable_timesync_service
+    end
+
+  private
+
+    # Configure Salt minion
+    def configure_salt_minion(master)
+      master_conf = ::Y2Caasp::CFA::MinionMasterConf.new
+
       begin
         master_conf.load
       rescue Errno::ENOENT
         log.info("The minion master.conf file does not exist, it will be created")
       end
-      log.info("The controller node for this worker role is: #{master}")
+
       # FIXME: the cobblersettings lense does not support dashes in the url
       # without single quotes, we need to use a custom lense for salt conf.
       # As Salt can use also 'url' just use in case of dashed.
       master_conf.master = master.include?("-") ? "'#{master}'" : master
```

Comment:
Maybe that's something that we should implement into `MinionMasterConf`. After all, it's just a workaround and this class should not know anything about that. @schubi2 already did something like that in [yast-configuration-management](https://github.com/yast/yast-configuration-management/commit/36dafacea174bb731fc4744305d5aee4fe9727e7).

And yes, this class (and the one used in yast-configuration-management) should be merged and moved to yast-yast2 (or just live in yast-configuration-management). But for the time being, that module does not exist for CaaSP.